### ISSUE TITLE
`handle_message_set_gps_global_origin`: use packet timestamp

### ISSUE THREAD

**Issue by obicons** [PX4/PX4-Autopilot#20018]
[`handle_message_hil_state_quaternion`](https://github.com/PX4/PX4-Autopilot/blob/dfdfbbfa9ce7de0d644b6ee11d54102a7a3bc486/src/modules/mavlink/mavlink_receiver.cpp#L2749) sets `timestamp_sample` to `hrt_absolute_time`. Instead, we should use `mavlink_timesync` to synchronize the packet's timestamp and use this value for `timestamp_sample`. I'll open a PR if we agree this is a better approach.

**Comment by dagar** [comment]
Yes you can do this everywhere that provides a timestamp, although there might be cases where we need to be careful with the fallback if/when time sync isn't provided.
